docs/refs/api/rtos/callbacks.md

```diff
@@ -169,25 +70,13 @@ int main() {
 }
 ```
 
-#### Combinatorial explosion damage control (aka why a separate class for Callbacks)
+#### A separate class for Callbacks)
 
-Unfortunately, supporting all of the standard C++ function types is difficult.
+C++ requires a large set of overloads to support all of the standard function types. It is unreasonable to expect a new library author to add all of these overloads to every function that could take in a callback.
```

Comment:
What about combining these two:
Unfortunately, supporting all of the standard C++ function types is difficult. C++ requires a large set of overloads to support all of the standard function types. It is unreasonable to expect a new library author to add all of these overloads to every function that could take in a callback.

Although now we're making claims without backing up the points. I'm worried that's just going to lead to more users underestimating the problems arount function argument overloads and roll their own hacks rather than seeing the value of the Callback class and adopting it.

The problems with incompatible Callback types only grows if we don't convince readers that staying in a single boat is the best thing to do.

Reply:
Although now we're making claims without backing up the points. I'm worried that's just going to lead to more users underestimating the problems arount function argument overloads and roll their own hacks rather than seeing the value of the Callback class and adopting it.

The problems with incompatible Callback types only grows if we don't convince readers that staying in a single boat is the best thing to do.